clients/vscode/src/Commands.ts

```diff
@@ -311,6 +312,18 @@ export class Commands {
           },
         },
       };
+      if (range) {
+        editLocation = {
+          uri: editor.document.uri.toString(),
+          range: {
+            start: { line: range.start.line, character: 0 },
+            end: {
+              line: range.end.line === 0 ? range.end.line : range.end.line + 1,
+              character: range.end.line === 0 ? Number.MAX_SAFE_INTEGER : 0,
```

Comment:
I believe the range process should aim to ensure that the range always covers full lines. However, the check here is `range.end.line` instead of `range.end.character`, which may be a mistake.

Additionally, this process should be same with Line 305-313, where you extract `editor.selection` to `editRange` and use `const editRange = range ? editor.selection` in order to reuse this process.